osu.Game/Database/BeatmapDatabase.cs

```diff
@@ -112,7 +111,17 @@ public void Import(params string[] paths)
                     }
                 }
 
-                Import(new[] { beatmapSet });
+                try
+                {
+                    Import(beatmapSet);
+                }
+                catch
+                {
+                }
+                finally
+                {
+                    File.Delete(p);

```

Comment:
Perhaps there should be a setting on whether to remove the beatmap file, as I'm sure that there are people who would like to keep the file and those who would like it to be removed. Or when importing many files the user could be prompted on whether to remove them or not.